### PR TITLE
Avatars/dApp Avatar Component

### LINKED ISSUE #20129: Avatars/dApp-Avatar
Design: https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=34075-53475&m=dev

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #20129 
### Summary

Implements the [Avatars/dApp-Avatar ](https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=34075-53475&m=dev)component

### Testing notes
Manual QA can be skipped since this PR only adds a quo component and doesn't introduce any changes outside Quo Preview.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
None

### Steps to test
Quo Preview -> Avatars -> dApp Avatar

status: ready

### CODE REVIEW

**Review by J-Son89 — APPROVED** [review]

**Review by BalogunofAfrica — APPROVED** [review]

File: src/quo/components/avatars/dapp_avatar/view.cljs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            [react-native.hole-view :as hole-view]
+            [schema.core :as schema]))
+
+(defn view-internal
+  [{:keys [context? image network-image container-style]}]
+  [rn/view
+   {:style               (merge style/container container-style)
+    :accessibility-label :dapp-avatar}
+   [hole-view/hole-view
+    {:key   (hash context?)
```
Comment (briansztamfater) [inline src/quo/components/avatars/dapp_avatar/view.cljs]: Same as here https://github.com/status-im/status-mobile/pull/20141#discussion_r1609847148, wdyt?

**Review by briansztamfater — APPROVED** [review]

**Review by ulisesmac — APPROVED** [review]

File: src/quo/components/avatars/dapp_avatar/view.cljs (hunk 1 above)
Comment (ulisesmac) [reply src/quo/components/avatars/dapp_avatar/view.cljs]: @ajayesivan isn't it enough if we only provide `context?` or maybe booleans arent a valid react key?

**Review by vkjr — APPROVED** [review]

File: src/quo/components/avatars/dapp_avatar/view.cljs (hunk 1 above)
Comment (ajayesivan) [reply src/quo/components/avatars/dapp_avatar/view.cljs]: @briansztamfater Changed the code to only add `key` prop for Android

Comment (ajayesivan) [reply src/quo/components/avatars/dapp_avatar/view.cljs]: @ulisesmac Yes we can use `boolean` value for `:key`. I have updated the code.

**Comment by ajayesivan** [comment]
Hi @Francesca-G, could you please do a design review for this PR?

Thanks!

**Review by Francesca-G — APPROVED** [review]
All good ✨

**Pull request merged**